<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Run GISTIC2 — gistic2 • blit</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Run GISTIC2 — gistic2"><meta property="og:description" content='The GISTIC module identifies regions of the genome that are
significantly amplified or deleted across a set of samples. Each aberration
is assigned a G-score that considers the amplitude of the aberration as well
as the frequency of its occurrence across samples. False Discovery Rate
q-values are then calculated for the aberrant regions, and regions with
q-values below a user-defined threshold are considered significant. For each
significant region, a "peak region" is identified, which is the part of the
aberrant region with greatest amplitude and frequency of alteration. In
addition, a "wide peak" is determined using a leave-one-out algorithm to
allow for errors in the boundaries in a single sample. The "wide peak"
boundaries are more robust for identifying the most likely gene targets in
the region. Each significantly aberrant region is also tested to determine
whether it results primarily from broad events (longer than half a chromosome
arm), focal events, or significant levels of both. The GISTIC module reports
the genomic locations and calculated q-values for the aberrant regions. It
identifies the samples that exhibit each significant amplification or
deletion, and it lists genes found in each "wide peak" region.'><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">blit</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/WangLabCSU/blit/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run GISTIC2</h1>
    <small class="dont-index">Source: <a href="https://github.com/WangLabCSU/blit/blob/HEAD/R/cmd-gistic2.R" class="external-link"><code>R/cmd-gistic2.R</code></a></small>
    <div class="hidden name"><code>gistic2.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The GISTIC module identifies regions of the genome that are
significantly amplified or deleted across a set of samples. Each aberration
is assigned a G-score that considers the amplitude of the aberration as well
as the frequency of its occurrence across samples. False Discovery Rate
q-values are then calculated for the aberrant regions, and regions with
q-values below a user-defined threshold are considered significant. For each
significant region, a "peak region" is identified, which is the part of the
aberrant region with greatest amplitude and frequency of alteration. In
addition, a "wide peak" is determined using a leave-one-out algorithm to
allow for errors in the boundaries in a single sample. The "wide peak"
boundaries are more robust for identifying the most likely gene targets in
the region. Each significantly aberrant region is also tested to determine
whether it results primarily from broad events (longer than half a chromosome
arm), focal events, or significant levels of both. The GISTIC module reports
the genomic locations and calculated q-values for the aberrant regions. It
identifies the samples that exhibit each significant amplification or
deletion, and it lists genes found in each "wide peak" region.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">gistic2</span><span class="op">(</span><span class="va">seg</span>, <span class="va">refgene</span>, <span class="va">...</span>, odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>, gistic2 <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-seg">seg<a class="anchor" aria-label="anchor" href="#arg-seg"></a></dt>
<dd><p>A data.frame of segmented data.</p></dd>


<dt id="arg-refgene">refgene<a class="anchor" aria-label="anchor" href="#arg-refgene"></a></dt>
<dd><p>Path to reference genome data input file (REQUIRED, see below
for file description).</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>&lt;<a href="https://rlang.r-lib.org/reference/dyn-dots.html" class="external-link">dynamic dots</a>&gt; Additional arguments passed to <code>gistic2</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="https://rdrr.io/r/base/shQuote.html" class="external-link">shQuote()</a></code>. Details see: <code>cmd_help(gistic2())</code>.</p></dd>


<dt id="arg-odir">odir<a class="anchor" aria-label="anchor" href="#arg-odir"></a></dt>
<dd><p>A string of path to the output directory.</p></dd>


<dt id="arg-gistic-">gistic2<a class="anchor" aria-label="anchor" href="#arg-gistic-"></a></dt>
<dd><p>A string of path to <code>gistic2</code> command.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A <code>command</code> object.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index">
<ul><li><p><a href="https://broadinstitute.github.io/gistic2/" class="external-link">https://broadinstitute.github.io/gistic2/</a></p></li>
<li><p><code><a href="cmd_wd.html">cmd_wd()</a></code>/<code><a href="cmd_wd.html">cmd_envvar()</a></code>/<code><a href="cmd_wd.html">cmd_envpath()</a></code>/<code><a href="cmd_wd.html">cmd_condaenv()</a></code></p></li>
<li><p><code><a href="cmd_on_start.html">cmd_on_start()</a></code>/<code><a href="cmd_on_start.html">cmd_on_exit()</a></code></p></li>
<li><p><code><a href="cmd_on_start.html">cmd_on_succeed()</a></code>/<code><a href="cmd_on_start.html">cmd_on_fail()</a></code></p></li>
<li><p><code><a href="cmd_parallel.html">cmd_parallel()</a></code></p></li>
</ul><p>Other <code>commands</code>:
<code><a href="allele_counter.html">allele_counter</a>()</code>,
<code><a href="bowtie2.html">bowtie2</a>()</code>,
<code><a href="bcftools.html">bcftools</a>()</code>,
<code><a href="cellranger.html">cellranger</a>()</code>,
<code><a href="conda.html">conda</a>()</code>,
<code><a href="fastp.html">fastp</a>()</code>,
<code><a href="fastq_pair.html">fastq_pair</a>()</code>,
<code><a href="kraken2.html">kraken2</a>()</code>,
<code><a href="kraken_tools.html">kraken_tools</a>()</code>,
<code><a href="perl.html">perl</a>()</code>,
<code><a href="pyscenic.html">pyscenic</a>()</code>,
<code><a href="python.html">python</a>()</code>,
<code><a href="samtools.html">samtools</a>()</code>,
<code><a href="seqkit.html">seqkit</a>()</code>,
<code><a href="trust4.html">trust4</a>()</code>,
<code><a href="varscan.html">varscan</a>()</code></p></div>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Yun Peng, Shixiang Wang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

